### PR TITLE
Add `ToastPresentation` component

### PR DESCRIPTION
**Pull request by mayank99**
Extracted the presentational part of `Toast` into a separate component so that it can be used with custom logic.

Closes # 

## Checklist

- [ ] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] ~~Add component features demo in Storybook (different stories)~~
- [x] ~~Approve test images for new stories~~
- [ ] Add an entry to the changelog for any new components and changes
- [ ] Add screenshots of the key elements of the component
- [x] Add any additional comments describing the features you've implemented

### CODE REVIEW

**Comment by mayank99** [comment]
@DanEastBentley Can you check if this will allow you to bring your own animation and rendering logic?

You can pull this branch and link itwinui-react locally.
https://github.com/iTwin/iTwinUI-react/blob/56154f5b6833f1a7774ddee59a8c755b8215c506/CONTRIBUTING.md#L149

You'll have to import from the file, because I don't think we're gonna export this in barrel (at least not in the current state).

**Comment by DanEastBentley** [comment]
Yes, I believe we will be able to use that. Thanks.

**Comment by DanEastBentley** [comment]
When will this be merged into master and into a build?

**Comment by mayank99** [comment]
I think this PR could be merged and released this week. It's pending review for things like naming (and whether an internal component like this should be advertised).

File: src/core/Toast/Toast.tsx [hunk 1]
```diff
@@ -150,32 +149,64 @@ export const Toast = (props: ToastProps) => {
...
-                  size='small'
-                  styleType='borderless'
-                  onClick={close}
-                  aria-label='Close'
-                >
-                  <SvgCloseSmall />
-                </IconButton>
-              )}
-            </div>
+            {ToastPresentation({
```
Comment (veekeys) [inline src/core/Toast/Toast.tsx:152]: Looks weird...Why not `<ToastPresentation />`?

File: src/core/Toast/Toast.tsx [hunk 2]
```diff
@@ -150,32 +149,64 @@ export const Toast = (props: ToastProps) => {
...
+export const ToastPresentation = (
+  props: Omit<ToastProps, 'duration' | 'id' | 'isVisible'> & CommonProps,
+) => {
+  const {
+    content,
+    category,
+    type = 'temporary',
+    link,
+    hasCloseButton,
+    onRemove,
```
Comment (veekeys) [inline src/core/Toast/Toast.tsx:180]: hmm.. discussable.
I would say it onClose then (cause its called only when you click the close 'x' button)

File: src/core/Toast/Toast.tsx [hunk 3]
```diff
@@ -150,32 +149,64 @@ export const Toast = (props: ToastProps) => {
...
     </Transition>
   );
 };
 
+/**
+ * The presentational part of a toast notification, without any animation or logic.
+ * @private
+ */
+export const ToastPresentation = (
+  props: Omit<ToastProps, 'duration' | 'id' | 'isVisible'> & CommonProps,
```
Comment (veekeys) [inline src/core/Toast/Toast.tsx:172]: well, we need `ToastPresentationProps` then...

File: src/core/Toast/Toast.tsx (hunk 1 above)
Comment (mayank99) [reply src/core/Toast/Toast.tsx]: Oh, all I did was a VSCode refactor. Fixed now

File: src/core/Toast/Toast.tsx (hunk 2 above)
Comment (mayank99) [reply src/core/Toast/Toast.tsx]: Changed to `onClose`. I didn't want to declare new props for this internal component but I guess it's fine 🤷‍♀️

File: src/core/Toast/Toast.tsx (hunk 3 above)
Comment (mayank99) [reply src/core/Toast/Toast.tsx]: Added

**Review by veekeys — APPROVED** [review]

**Comment by mayank99** [comment]
@DanEastBentley This is now available in 1.10.0 🎉

**Pull request merged**